pkg/kubelet/kubelet.go

```diff
@@ -428,7 +428,7 @@ func NewMainKubelet(
 			imageBackOff,
 			serializeImagePulls,
 			enableCustomMetrics,
-			klet.hairpinMode == componentconfig.HairpinVeth,
+			klet.hairpinMode == componentconfig.HairpinVeth && networkPluginName != "kubenet",
```

Comment:
please add a comment, something like:

kubenet is the Kubernetes network plugin that wraps CNI's bridge plugin. The bridge plugin knows how to set the hairpin veth flag on interfaces it creates for containers, so we assume kubenet is piping the flag right through, and tell the container runtime to back away from setting it. If the kubelet is started with any other plugin, we can't be sure it handles the hairpin case, so we instruct the docker runtime to set the flag instead.